errai-codegen/src/main/java/org/jboss/errai/codegen/util/GenUtil.java

```diff
@@ -480,10 +480,15 @@ else if (arguments.length == 0 && parmTypes.length == 0) {
 
           score = scoreMethods(arguments, parmTypes, isVarArgs);
 
+
           if (score != 0 && score > bestScore) {
             bestCandidate = meth;
             bestScore = score;
           }
+
+          if (score != 0 && score == bestScore && bestCandidate.isVarArgs()) {
```

Comment:
Could you add a comment here explaining what corner case this is? Is this so we always prefer a var-args method over non-var-args when they are otherwise equally valid matches?

Reply:
It's the contrary of that. We always prefer non-varargs methods over varargs ones. So at this point, when the bestCandidate is vararg, we switch. I'll add the comment.

Reply:
Hehe yes I got confused there. All the more reason for the comment ;)